<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrainingWidget</class>
 <widget class="QWidget" name="TrainingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>395</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QPushButton" name="StartTrainingButton">
     <property name="text">
      <string>Start Training</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="ParametersGridLayout">
     <item row="0" column="1">
      <widget class="QLineEdit" name="ReplayMemorySizeInput">
       <property name="text">
        <string>200000</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="ReplayMemorySizeLabel">
       <property name="text">
        <string>ReplaMemorySize: </string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="UseDrawsLabel">
       <property name="text">
        <string>UseDraws: </string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="UseDrawsYesRadio">
         <property name="text">
          <string>Yes</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">UseDrawsButtonGrp</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="UseDrawsNoRadio">
         <property name="text">
          <string>No</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">UseDrawsButtonGrp</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QRadioButton" name="RestrictGameLengthYesRadio">
         <property name="text">
          <string>Yes</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">RestrictGameLengthButtonGrp</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="RestrictGameLengthNoRadio">
         <property name="text">
          <string>No</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">RestrictGameLengthButtonGrp</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="RestrictGameLengthLabel">
       <property name="text">
        <string>Restrict Game Length: </string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="MaxGameLengthLabel">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Max Game Length: </string>
       </property>
       <property name="indent">
        <number>-1</number>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="MaxGameLengthInput">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QLineEdit" name="RandomizedMoveCountInput">
       <property name="text">
        <string>10</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="LearningRateLabel">
       <property name="text">
        <string>Learning Rate: </string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="SelfPlayMCTSCountInput">
       <property name="text">
        <string>100</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QLineEdit" name="LearningRateInput">
       <property name="text">
        <string>0.001</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="SaveIterationCountLabel">
       <property name="text">
        <string>Save After Iterations: </string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLineEdit" name="SelfPlayGamesCountInput">
       <property name="text">
        <string>1000</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="SelfPlayMCTSLabel">
       <property name="text">
        <string>Self-Play MCTS Count: </string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="TrainingIterationsLabel">
       <property name="text">
        <string>Training Iterations: </string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLineEdit" name="SaveIterationCountInput">
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="TrainingIterationsInput">
       <property name="text">
        <string>999999</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="RandomizedMoveCountLabel">
       <property name="text">
        <string>Randomized Move Count:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="SelPlayGamesCountLabel">
       <property name="text">
        <string>Self-Play Games per Iteration: </string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLineEdit" name="TrainingBatchSizeInput">
       <property name="text">
        <string>100</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="TrainingBatchSizeLabel">
       <property name="text">
        <string>Training Batch Size: </string>
       </property>
      </widget>
     </item>
     <item row="11" column="0">
      <widget class="QLabel" name="FilterCountLabel">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>NN-Filter Count: </string>
       </property>
      </widget>
     </item>
     <item row="11" column="1">
      <widget class="QLineEdit" name="FilterCountInput">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>512</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="12" column="0">
      <widget class="QLabel" name="DeviceLabel">
       <property name="text">
        <string>Device: </string>
       </property>
      </widget>
     </item>
     <item row="12" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QRadioButton" name="CPURadio">
         <property name="text">
          <string>CPU</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">DeviceButtonGrp</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="GPURadio">
         <property name="text">
          <string>GPU</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">DeviceButtonGrp</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
     <item row="13" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="CpuThreadsLabel">
         <property name="text">
          <string>CPU-Threads: </string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="13" column="1">
      <widget class="QLineEdit" name="CpuThreadsInput">
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLineEdit" name="Note">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Note: The Current Iteration is printed in the Console</string>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="RestrictGameLengthButtonGrp"/>
  <buttongroup name="DeviceButtonGrp"/>
  <buttongroup name="UseDrawsButtonGrp"/>
 </buttongroups>
</ui>
